File: Data Structures & Algorithms/plus-one/submission-1.go
func plusOne(digits []int) []int {
    n := len(digits)
	digits[n-1] += 1
	carry := 0
	for i := n - 1; i >= 0; i-- {
		digits[i] += carry
		carry = digits[i] / 10
		digits[i] %= 10
	}
	if carry == 0 {
		return digits
	}
	res := make([]int, n + 1)
	res[0] = 1
	for i := 1; i < n + 1; i++ {
		res[i] = digits[i-1]
	}
	return res
}

File: Data Structures & Algorithms/combinations-of-a-phone-number/submission-1.cpp
class Solution {
private:
    vector<string> chars_map = { 
        "", "", "abc", "def", "ghi",
        "jkl", "mno", "pqrs", "tuv", "wxyz"
    };
    vector<string> res;
public:
    void back_tracking(int i, string curr_str, string& digits) {
        if (i >= digits.length()) {
            res.push_back(curr_str);
            return;
        }
        string chars = chars_map[digits[i] - '0'];
        for (char c : chars) {
            back_tracking(i + 1, curr_str + c, digits);
        }
    }

    vector<string> letterCombinations(string digits) {
        if (digits.empty()) return res;
        back_tracking(0, "", digits);
        return res;
    }
};